crates/ethereum/consensus/src/validation.rs

```diff
@@ -172,18 +172,18 @@ mod tests {
         let expected_receipts_root = B256::random();
         let expected_logs_bloom = calculated_logs_bloom;
 
-        assert_eq!(
+        assert!(matches!(
             compare_receipts_root_and_logs_bloom(
                 calculated_receipts_root,
                 calculated_logs_bloom,
                 expected_receipts_root,
                 expected_logs_bloom
-            ),
-            Err(ConsensusError::BodyReceiptRootDiff(
-                GotExpected { got: calculated_receipts_root, expected: expected_receipts_root }
-                    .into()
-            ))
-        );
+            ).unwrap_err(),
+            ConsensusError::BodyReceiptRootDiff(err)
+                if matches!(*err, GotExpected { got, expected }
+                    if got == calculated_receipts_root && expected == expected_receipts_root
+                )
+        ));
```

Comment:
```suggestion
            ConsensusError::BodyReceiptRootDiff(diff) if
                diff.got == calculated_receipts_root && diff.expected == expected_receipts_root
        ));
```
prefer the simplified syntax from above!